lib/Controller/User.php

```diff
@@ -748,6 +956,49 @@ public function changePassword()
             'passwordUpdate' => true
         ]);
 
+        $user->isPasswordChangeRequired = 0;
+        $user->save();
+
+        // Return
+        $this->getState()->hydrate([
+            'message' => __('Password Changed'),
+            'id' => $user->userId,
+            'data' => $user
+        ]);
+    }
+
+    /**
+     * Force User Password Change
+     */
+    public function forceChangePasswordPage()
+    {
+        $this->getState()->template = 'user-force-change-password-page';
+    }
+
+    /**
+     * Force change my Password
+     */
+    public function forceChangePassword()
+    {
+        // Save the user
+        $user = $this->getUser();
+        $newPassword = $this->getSanitizer()->getString('newPassword');
+        $retypeNewPassword = $this->getSanitizer()->getString('retypeNewPassword');
+
+        if ($newPassword == null || $retypeNewPassword == '')
+            throw new \InvalidArgumentException(__('Please enter the password'));
```

Comment:
Can you change your InvalidArgumentExceptions to the one that extends from XiboException?